ccd-definitions/src/main/java/uk/gov/hmcts/reform/divorce/ccd/model/CaseData.java

```diff
@@ -169,4 +178,128 @@
     )
     private YesOrNo jurisdictionBothLastHabituallyResident;
 
+
+    @JsonProperty("D8PetitionerNameDifferentToMarriageCert")
+    @CCD(
+        label = "Has the petitioner changed their name since they got married?",
+        hint = "Is the respondent’s current name different to their married name or the name shown on their marriage certificate?"
+    )
+    private YesOrNo d8PetitionerNameDifferentToMarriageCert;
+
+    @JsonProperty("D8PetitionerNameChangedHow")
+    @CCD(
+        label = "How did the petitioner change their name?"
+    )
+    private ChangedNameHow d8PetitionerNameChangedHow;
+
+    @JsonProperty("D8PetitionerNameChangedHowOtherDetails")
+    @CCD(
+        label = "Provide details of how they changed their name",
+        type = TextArea
+    )
+    private String d8PetitionerNameChangedHowOtherDetails;
+
+    @JsonProperty("D8DivorceWho")
+    @CCD(
+        label = "Who is petitioner divorcing?",
+        hint = "Husband or Wife?",
+        type = FixedList,
+        typeParameter = "WhoDivorcing"
+    )
+    private WhoDivorcing d8DivorceWho;
+
+    @JsonProperty("D8DerivedPetitionerHomeAddress")
+    @CCD(
+        label = "The Petitioner's home address",
+        type = TextArea
+    )
+    private String d8DerivedPetitionerHomeAddress;
+
+    @JsonProperty("D8PetitionerPhoneNumber")
+    @CCD(
+        label = "Petitioner's phone number",
+        regex = "^[0-9 +().-]{9,}$"
+    )
+    private String d8PetitionerPhoneNumber;
+
+    @JsonProperty("D8PetitionerContactDetailsConfidential")
+    @CCD(
+        label = "Keep the petitioner's contact details private from the respondent?",
+        type = FixedList,
+        typeParameter = "ConfidentialAddress"
+    )
+    private ConfidentialAddress d8PetitionerContactDetailsConfidential;
```

Comment:
how does hiding of address works ? Should we control it via role access ?

Reply:
I believe the current approach is to use some logic in a show condition. I do wonder why they didn't do it with permissions